great_expectations/data_context/store/database_store_backend.py

```diff
@@ -55,12 +60,31 @@ def __init__(
                 "DatabaseStoreBackend requires use of a fixed-length-key"
             )
 
-        drivername = credentials.pop("drivername")
-        schema = credentials.pop("schema", None)
-        options = URL(drivername, **credentials)
-        self.engine = create_engine(options)
+        self._schema_name = None
+        self._credentials = credentials
+        self._connection_string = connection_string
+        self._url = url
+
+        if engine is not None:
+            if credentials is not None:
+                logger.warning(
+                    "Both credentials and engine were provided during initialization of SqlAlchemyExecutionEngine. "
+                    "Ignoring credentials."
+                )
+            self.engine = engine
+        elif credentials is not None:
+            self.engine = self._build_engine(credentials=credentials, **kwargs)
+        elif connection_string is not None:
+            self.engine = sa.create_engine(connection_string, **kwargs)
+        elif url is not None:
+            self.drivername = urlparse(url).scheme
+            self.engine = sa.create_engine(url, **kwargs)
+        else:
+            raise ge_exceptions.InvalidConfigError(
+                "Credentials, url, connection_string, or an engine are required for a DatabaseStoreBackend."
+            )
 
-        meta = MetaData(schema=schema)
+        meta = MetaData(schema=self._schema_name)
```

Comment:
`self._schema_name` can be assigned by the `_build_engine()` function, which has run by the time the code gets to this line